src/com/amazon/ion/impl/IonReaderBinaryIncremental.java

```diff
@@ -0,0 +1,1932 @@
+        private final SymbolTable system = SharedSymbolTable.getSystemSymbolTable(majorVersion);
+
+        // The max ID of this local symbol table.
+        private final int maxId;
+
+        // The max ID that maps to a shared symbol table imported by this local symbol table.
+        private final int importedTablesMaxId;
+
+        // The shared symbol tables imported by this local symbol table.
+        private final SymbolTable[] importedTables;
+
+        // Map representation of this symbol table. Keys are symbol text; values are the lowest symbol ID that maps
+        // to that text.
+        final Map<String, Integer> mapView;
+
+        // List representation of this symbol table, indexed by symbol ID.
+        final List<String> listView;
+
+        LocalSymbolTableSnapshot() {
+            maxId = numberOfSymbols - 1;
+            importedTablesMaxId = importMaxId;
+            // Map with initial size the number of symbols and load factor 1, meaning it must be full before growing.
+            // It is not expected to grow.
+            listView = new ArrayList<String>(symbols.subList(0, numberOfSymbols));
+            mapView = new HashMap<String, Integer>(numberOfSymbols, 1);
+            for (int i = 0; i < numberOfSymbols; i++) {
+                String symbol = listView.get(i);
+                if (symbol != null) {
+                    mapView.put(symbol, i);
+                }
+            }
+            importedTables = new SymbolTable[imports.size()];
+            int i = 0;
+            for (SymbolTable importedTable : imports.values()) {
+                importedTables[i] = importedTable;
+                i++;
+            }
+        }
+
+        @Override
+        public String getName() {
+            return null;
+        }
+
+        @Override
+        public int getVersion() {
+            return 0;
+        }
+
+        @Override
+        public boolean isLocalTable() {
+            return true;
+        }
+
+        @Override
+        public boolean isSharedTable() {
+            return false;
+        }
+
+        @Override
+        public boolean isSubstitute() {
+            return false;
+        }
+
+        @Override
+        public boolean isSystemTable() {
+            return false;
+        }
+
+        @Override
+        public SymbolTable getSystemSymbolTable() {
+            return system;
+        }
+
+        @Override
+        public String getIonVersionId() {
+            return system.getIonVersionId();
+        }
+
+        @Override
+        public SymbolTable[] getImportedTables() {
+            return importedTables;
+        }
+
+        @Override
+        public int getImportedMaxId() {
+            return importedTablesMaxId;
+        }
+
+        @Override
+        public SymbolToken find(String text) {
+            Integer sid = mapView.get(text);
+            if (sid == null) {
+                return null;
+            }
+            return new SymbolTokenImpl(text, sid, null);
```

Comment:
Agreed

Reply:
That's a solid choice- a comment saying the same might be worthwhile and just because it informs about this specific choice but because it illuminates the intent and reasoning behind this and other choices in the code in a way that will help your posterity make better choices. 
[Code Only Says What it Does](https://brooker.co.za/blog/2020/06/23/code.html)

Reply:
Done.